src/pages/MainPage.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { ZaboEmbed } from 'zabo-embed';
+import { appBoundClassNames as classNames } from '../common/boundClassNames';
+import Footer from '../common/guideline/components/Footer';
+import TodaysTimetableSection from '../components/sections/main/TodaysTimetableSection';
+import AcademicScheduleSection from '../components/sections/main/AcademicScheduleSection';
+import RelatedCourseFeedSection from '../components/sections/main/RelatedCourseFeedSection';
+import LatestReviewSection from '../components/sections/main/LatestReviewSection';
+import FamousMajorReviewFeedSection from '../components/sections/main/FamousMajorReviewFeedSection';
+import FamousHumanityReviewFeedSection from '../components/sections/main/FamousHumanityReviewFeedSection';
+import RankedReviewFeedSection from '../components/sections/main/RankedReviewFeedSection';
+import ReviewWriteFeedSection from '../components/sections/main/ReviewWriteFeedSection';
+import MainSearchSection from '../components/sections/main/MainSearchSection';
+import NoticeSection from '../components/sections/main/NoticeSection';
+import RateFeedSection from '../components/sections/main/RateFeedSection';
+import { rootReducer } from '@/App';
+import { range, throttle } from 'lodash';
+import { useFeed, useNotices } from '@/queries/main';
+
+type RootState = ReturnType<typeof rootReducer>;
+
+// TODO: Feed 의 타입은 이후에 API 타입 패키지로 바꿀 예정
+// TODO: 스크롤 감지 interserct observer 로 바꾸기
+const MainPage: React.FC = () => {
+  const contentRef = useRef<HTMLDivElement | null>(null);
+
+  const { user } = useSelector((state: RootState) => state.common.user);
```

Comment:
위에서 @withSang 님께서도 말씀해주셨는데, tanstack에서 관리하고 있는 user session을 사용하려면 이렇게만 작성해주시면 됩니다. 세션이 없을 때 처리가 (아마) 되어 있지 않아서 이번 PR에서 반영할지는 선택 사항.. 
```suggestion
  const { data: user } = useSessionInfo();
```

Reply:
머지는 안 되어 있는데, `migration@sections/timetable` 브랜치도 참고할 만할 거예요. `TimetableTabs.tsx`, `timetable.ts`에서 각각 `userSessionInfo`를 사용하고 있습니다.

Reply:
https://github.com/sparcs-kaist/otlplus-web/pull/83#discussion_r1764936359
위 스레드 참고.! 
pr 분리해서 에러 처리 + 새로 작성 컴포넌트에 한해서 tanstack 으로 세션 가져오기 이렇게 바꾸면 될 거 같아용